test/helpers/helpers.js

```diff
@@ -37,6 +38,21 @@ beforeEach(() => {
           return result;
         }
       };
-    }
+    },
+    toDeepEqual: () => ({
+      compare: async(spec, expected) => {
+        const result = {};
+        const actual = marked(spec.markdown, spec.options);
+        result.pass = assert.deepStrictEqual(expected, actual) === undefined;
+
+        if (result.pass) {
+          result.message = `${spec.markdown}\n------\n\nExpected: Should Fail`;
+        } else {
+          const diff = await htmlDiffer.firstDiff(actual, expected);
```

Comment:
I don't think htmlDiffer will be useful here since it might not find a difference if it is just in whitespace.

we will probably need to loop over the strings to find the index where they change and show 30 chars on each side.

Reply:
This is solved by using `assert.deepStrictEqual` in case of discrepancy, he shall provide comprehensive information on the discrepancy 
![Screenshot 2020-04-17 at 09 25 14](https://user-images.githubusercontent.com/2789192/79538725-91052900-808d-11ea-92c7-b6b0cebc0b31.png)

I updated this method, removing all unnecessary